de.metas.handlingunits.client/src/main/java/de/metas/handlingunits/client/terminal/editor/model/impl/MovementsAnyWarehouseModel.java

```diff
@@ -87,4 +85,40 @@ protected void createMovements()
 
 	}
 
+	/**
+	 * Get the warehouses existing for the hus' organization , excluding those which currently contain the HUs given as parameters
+	 * 
+	 * @param hus
+	 * @return
+	 */
+	public static List<org.compiere.model.I_M_Warehouse> getWarehousesUnlessOfHUs(final List<I_M_HU> hus)
+	{
+		if (hus.isEmpty())
+		{
+			// should never happen
+			return Collections.emptyList();
+		}
+
+		// used for deciding the org and context
+		final I_M_HU firstHU = hus.get(0);
+
+		final int orgId = firstHU.getAD_Org_ID();
+		final Properties ctx = InterfaceWrapperHelper.getCtx(firstHU);
+
```

Comment:
What we think we would have done here is something like:
* extend IWarehouseDAO with a method such as `retrieve(WarehouseQuery)` and WarehouseQuery would have `getAD_Org_ID()` and `getExcludedWarehouseIds()` (maybe more)
* and then *first` then get my IDs from those streams and pass them to the warehouseDAO..instead of first getting all warehouses and then removing some...

But in order of getting the job done now, this is also OK